packages/@o3r/design/src/core/design-token/renderers/css/design-token-value.renderers.ts

```diff
@@ -0,0 +1,41 @@
+import type { DesignTokenVariableStructure, TokenKeyRenderer, TokenValueRenderer } from '../../parsers/design-token-parser.interface';
+import { shouldDefineVariableValueFromOtterInfo } from '../design-token.renderer.helpers';
+
+interface CssTokenValueRendererOptions {
+  /**
+   * Determine if the variable should be rendered, based on Otter Extension information
+   * @default {@see shouldDefineVariableValueFromOtterInfo}
+   */
+  shouldDefineVariable?: (variable: DesignTokenVariableStructure) => boolean;
+
+  /**
+   * Renderer the name of the CSS Variable (without initial --)
+   */
+  tokenVariableNameRenderer?: TokenKeyRenderer;
+}
+
+/**
+ * Retrieve the Design Token value renderer
+ * @param options
+ */
+export const getCssTokenValueRenderer = (options?: CssTokenValueRendererOptions): TokenValueRenderer => {
+  const isVariableToDefined = options?.shouldDefineVariable || shouldDefineVariableValueFromOtterInfo;
+  const tokenVariableNameRenderer = options?.tokenVariableNameRenderer;
+
+  const referenceRenderer = (variable: DesignTokenVariableStructure, variableSet: Map<string, DesignTokenVariableStructure>): string => {
+    if (isVariableToDefined(variable)) {
+      return `var(--${variable.getKey(tokenVariableNameRenderer)})`;
```

Comment:
If variable should be defined, why is it not done for this clause but the other?

Reply:
not sure to understand your question :S.
If `variable` should be defined, it means we can refer to it without fallback value (because we guaranty that it exists).
In the opposite, as we can not guaranty it, we should always provide a fallback value.
Did it answer to your question?

Reply:
Yes, I understood the other way round